src/jit/lclvars.cpp

```diff
@@ -1563,9 +1565,16 @@ void Compiler::lvaCanPromoteStructType(CORINFO_CLASS_HANDLE    typeHnd,
                     return;
                 }
 
-                // Non-primitive struct field.
-                // Try to promote structs of single field of scalar types aligned at their
-                // natural boundary.
+                // Try to promote byref-like structs with a single field of scalar type that is aligned at its
+                // type's natural boundary.
+
+                // Do not promote if the single field is not byref-like.
+                const DWORD fieldFlags       = info.compCompHnd->getClassAttribs(pFieldInfo->fldTypeHnd);
+                const bool  fieldIsByRefLike = (fieldFlags & CORINFO_FLG_CONTAINS_STACK_PTR) != 0;
+                if (!fieldIsByRefLike)
+                {
+                    return;
+                }
 
```

Comment:
I think I understand the reason to be conservative here, but this will also disable promoting a field that's a struct containing a single ptr-sized int. That seems unnecessarily pessimizing. Couldn't you just check below that `fieldVarType` is not `TYP_REF`?